packages/text/src/TextMetrics.js

```diff
@@ -699,7 +699,21 @@ export default class TextMetrics
  * @private
  */
 
-const canvas = document.createElement('canvas');
+const canvas = (() =>
```

Comment:
Why not something more simple? Is there something more subtle here about OffscreenCanvas that I'm not understanding that requires a try/catch?
```suggestion
const canvas = window.OffscreenCanvas ?
    new OffscreenCanvas(0, 0):
    document.createElement('canvas');
```

Reply:
Yes. OffscreenCanvas can be implemented with only WebGL support (currently on Firefox). So we need to `getContext("2d")` and try/catch to see if 2D OffscreenCanvas is supported.